SignalServiceKit/src/Messages/OWSMessageReceiver.m

```diff
@@ -333,6 +333,28 @@ + (void)syncRegisterDatabaseExtension:(YapDatabase *)database
 
 - (void)handleReceivedEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
 {
+    // Drop any too-large messages on the floor. Well behaving clients should never send them.
+    NSUInteger kMaxEnvelopeByteCount = 250 * 1024;
+    if (envelope.serializedSize > kMaxEnvelopeByteCount) {
+        OWSProdErrorWParams(@"message_receiver_error_oversize_message", ^{
+            return (@{
+                @"message_size" : @([OWSAnalytics orderOfMagnitudeOf:(long)envelope.serializedSize]),
+            });
+        });
+        return;
+    }
+
+    // Take note of any messages larger than we expect, but still process them.
+    // This likely indicates a misbehaving sending client.
+    NSUInteger kLargeEnvelopeWarningByteCount = 25 * 1024;
```

Comment:
I'm guessing that the largest valid envelope should be some small multiple of the largest possible text message.

```
        const NSUInteger kOversizeTextMessageSizeThreshold = 16 * 1024;
```

I'm guessing that a 16k text message would yield an envelope smaller than 25k, but we should verify that?